tests/integration/test_load_and_run_agent.py

```diff
@@ -219,5 +218,8 @@ def write_file(text: str) -> None:
             html_output = agent._exporter.console.export_html(inline_styles=True)  # type: ignore[union-attr]
             with open(f"{trace_path}_trace.html", "w", encoding="utf-8") as f:
                 f.write(html_output.replace("<!DOCTYPE html>", ""))
+
+        agent.exit()
+        assert_eval(agent_trace)
```

Comment:
Is there a way that we can make it so that exit() is automatically called on garbage collection? I'm still wishing that we didn't need to manually call agent.exit().

Reply:
Yeah but if it's not called at all, what happens when the tracer is garbage collected (whenever that might be). Does the uninstrument happen? I just want to make sure the cleanup does happen and we don't accidentally fail to fix some sort of global state. I have no idea if this is possible in the tracing design.